### PR TITLE
Add handle file upload when click

### PR DESCRIPTION
**Pull request by GGolfz**
PR for #3 

This pull request adds some changes to the existing codebase. It introduces a new hidden file input field to handle file uploads, along with a new method called `processFile` to handle the file processing.

The old `handleDrop` function has been refactored to pass the `FileList` object to the new `processFile` method, instead of handling the file processing logic directly. This change makes the code more modular and easier to maintain.

### CODE REVIEW

File: apps/visualizer/src/pages/home.tsx [hunk 1]
```diff
@@ -10,31 +10,51 @@ export default function Home() {
 
   const [errorMessage, setError] = createSignal<string | null>(null)
 
-  async function handleDrop(e: DragEvent) {
+  function handleDrop(e: DragEvent) {
     e.preventDefault()
+    processFile(e.dataTransfer?.files || new FileList())
```
Comment (heypoom) [inline apps/visualizer/src/pages/home.tsx:15]: I'm curious what is the difference between `FileList` and `[]` krub

File: apps/visualizer/src/pages/home.tsx [hunk 2]
```diff
@@ -10,31 +10,51 @@ export default function Home() {
...
 
-    const transactions = (
-      await Promise.all(files.map(processStatementFile))
-    ).flat()
+  const headers = ["Amount", "Payment Date"]
 
-    setTransactions(transactions)
+  function handleClickUploadFile(e: MouseEvent) {
+    e.preventDefault()
+    document.getElementById("statement-input")?.click()
```
Comment (heypoom) [inline apps/visualizer/src/pages/home.tsx:26]: Recommendation is to refactor this to attach refs with `ref={input}`, so it does not rely on getElementById krub

https://www.solidjs.com/tutorial/bindings_refs

**Review by heypoom — APPROVED** [review]
Looks great! Added a recommendation on refs.

File: apps/visualizer/src/pages/home.tsx (hunk 1 above)
Comment (GGolfz) [reply apps/visualizer/src/pages/home.tsx:15]: The type of `[]` is `undefined[]`, which might raise a Type incompatible error.

File: apps/visualizer/src/pages/home.tsx (hunk 2 above)
Comment (GGolfz) [reply apps/visualizer/src/pages/home.tsx:26]: Thanks krub! Almost forgot about `ref `

**Pull request merged**